ros/src/tl_detector/tl_detector.py

```diff
@@ -272,6 +268,22 @@ def get_light_state(self, light):
         # Get classification
         return self.light_classifier.get_classification(img_traffic_light)
 
+    def get_tl_waypoints(self):
+        """ Converts list self.config['light_positions'] with trafic
+        light positions to get_closest_waypoint array with PoseStamped
+        traffic light waypoints
+
+        """
+
+        for light_position in self.config['light_positions']:
+            tl_waypoint = PoseStamped()
+            tl_waypoint.pose.position.x = light_position[0]
+            tl_waypoint.pose.position.y = light_position[1]
+            tl_waypoint.pose.position.z = 0
+            self.tl_waypoints.append(
+                self.get_closest_waypoint(tl_waypoint.pose)
```

Comment:
Should we worry about choosing a waypoing past the traffic light? Then we will command the car to have velocity = 0.0 at a point past the traffic light.

Reply:
I have already considered that, only traffic lights in front of the car will be selected.

Reply:
What I mean is that for a given `Pose`, the function `get_closest_waypoint` will always return the closest waypoint to the `Pose`, no matter if that waypoint is _after_ the `Pose` or _before_ the `Pose`. Previously we discussed that the waypoints are very dense so it doesn't really matter much.

Now, I realize that if you do `get_closest_waypoint(tl_waypoint.pose)`, it can happen that the chosen waypoint is _after_ the traffic light. And the DBW node is going to fully stop the car at that specific waypoint - in other words the car can potentially stop _after_ crossing the red light.

Reply:
When I tested it in the simulator traffic light detection always ended exactly as the car entered the crossing, as the waypoints exact location is always start of the crossing. 

I'm not sure how this is going to work in real life.

Reply:
We can add one waypoint extra to car position to be sure.

Reply:
Ok! Let's merge this so we can test and refine later